roxie/ccd/ccdmain.cpp

```diff
@@ -1336,6 +1348,15 @@ int CCD_API roxie_main(int argc, const char *argv[], const char * defaultYaml)
             setNodeFetchThresholdNs(topology->getPropInt64("@nodeFetchThresholdNs"));
         setIndexWarningThresholds(topology);
 
+        //Enabling the localNVMeCache will also by default enable remote file related optimizations
+        offset_t localNVMeCacheSize = topology->getPropInt64("@localNVMeCacheSize", 0);
+        bool usingRemoteStorage = (localNVMeCacheSize != 0);
+
+        unsigned inplaceSizeFactor = topology->getPropInt("@inplaceSizeFactor", usingRemoteStorage ? 0 : 100);
+        unsigned lz4SpeedFactor = topology->getPropInt("@lz4SpeedFactor", usingRemoteStorage ? 0 : 600);             // If lz4 is 5x faster then value should be 500
+        unsigned zStdSpeedFactor = topology->getPropInt("@zStdSpeedFactor", usingRemoteStorage ? 0 : 350);
+        setIndexScaling(inplaceSizeFactor, lz4SpeedFactor, zStdSpeedFactor);
+
```

Comment:
It might be nice if we could set these speed Factors in the query xml (ie via context).